examples/demo-stf/src/genesis_config.rs

```diff
@@ -32,63 +43,58 @@ pub const DEMO_TOKEN_NAME: &str = "sov-demo-token";
     sequencer_da_address: Da::Address,
+    genesis_paths: &GenesisPaths<P>,
     #[cfg(feature = "experimental")] eth_signers: Vec<reth_primitives::Address>,
 ) -> GenesisConfig<C, Da> {
-    let token_deployer: PrivateKeyAndAddress<C> = read_private_key();
-
     create_genesis_config(
-        token_deployer.address.clone(),
         sequencer_da_address,
+        genesis_paths,
         #[cfg(feature = "experimental")]
         eth_signers,
     )
     .expect("Unable to read genesis configuration")
 }
 
-fn create_genesis_config<C: Context, Da: DaSpec>(
-    sequencer_address: C::Address,
-    sequencer_da_address: Da::Address,
+fn create_genesis_config<C: Context, Da: DaSpec, P: AsRef<Path>>(
+    seq_da_address: Da::Address,
+    genesis_paths: &GenesisPaths<P>,
     #[cfg(feature = "experimental")] eth_signers: Vec<reth_primitives::Address>,
 ) -> anyhow::Result<GenesisConfig<C, Da>> {
-    // This path will be injected as a parameter: #872
-    let bank_genesis_path = "../test-data/genesis/bank.json";
-    let bank_config: BankConfig<C> = read_json_file(bank_genesis_path)?;
-    // This will be read from a file: #872
-    let token_address = sov_bank::get_genesis_token_address::<C>(
-        &bank_config.tokens[0].token_name,
-        bank_config.tokens[0].salt,
-    );
+    let bank_config: BankConfig<C> = read_json_file(&genesis_paths.bank_genesis_path)?;
 
-    // This will be read from a file: #872
-    let sequencer_registry_config = sov_sequencer_registry::SequencerConfig {
-        seq_rollup_address: sequencer_address,
-        seq_da_address: sequencer_da_address,
-        coins_to_lock: sov_bank::Coins {
-            amount: LOCKED_AMOUNT,
-            token_address,
-        },
-        is_preferred_sequencer: true,
-    };
+    let mut sequencer_registry_config: SequencerConfig<C, Da> =
+        read_json_file(&genesis_paths.sequencer_genesis_path)?;
 
-    // This path will be injected as a parameter: #872
-    let value_setter_genesis_path = "../test-data/genesis/value_setter.json";
-    let value_setter_config: ValueSetterConfig<C> = read_json_file(value_setter_genesis_path)?;
+    // The `seq_da_address` is overridden with the value from rollup binary.
+    sequencer_registry_config.seq_da_address = seq_da_address;
 
-    let accounts_genesis_path = "../test-data/genesis/accounts.json";
-    let accounts_config: AccountConfig<C> = read_json_file(accounts_genesis_path)?;
+    // Sanity check: `token_address` in `sequencer_registry_config` match `token_address` from the bank module.
+    {
+        let token_address = sov_bank::get_genesis_token_address::<C>(
+            &bank_config.tokens[0].token_name,
+            bank_config.tokens[0].salt,
+        );
 
-    let nft_config: NonFungibleTokenConfig = NonFungibleTokenConfig {};
+        assert_eq!(
+            sequencer_registry_config.coins_to_lock.token_address,
+            token_address
+        );
```

Comment:
This crash results from user-provided configuration values instead of a logic error in the code, so it should probably be an `anyhow::Error`.

Reply:
Fixed by https://github.com/Sovereign-Labs/sovereign-sdk/pull/972/commits/cb8932ccdb122f2e71ea97a5229148b7835753c7.